src/trans.c

```diff
@@ -1500,16 +1500,23 @@ Obj FuncON_KERNEL_ANTI_ACTION (Obj self, Obj ker, Obj f, Obj n) {
   UInt    deg, i, j, rank, len;
   Obj     out;
 
-  if (INT_INTOBJ(ELM_LIST(ker, LEN_LIST(ker))) == 0) {
+  assert(IS_LIST(ker) && IS_INTOBJ(n));
+
+  len = LEN_LIST(ker);
+  if (len == 1 && INT_INTOBJ(ELM_LIST(ker, 1)) == 0) {
```

Comment:
This was only ever meant to apply to the list `[0]` and was just wrong before. I replaced `LEN_LIST(ker)` by `1` here for the same reason as your next two comments.